app/src/main/kotlin/com/rockspin/bargainbits/ui/deals/DealListActivity.kt

```diff
@@ -0,0 +1,136 @@
+import com.rockspin.bargainbits.utils.Feedback
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.rxkotlin.merge
+import io.reactivex.subjects.PublishSubject
+import org.codechimp.apprater.AppRater
+import javax.inject.Inject
+
+class DealListActivity : BaseActivity(), StoreFilterDialogFragment.OnDialogClosedListener {
+
+    @Inject lateinit var adapter: DealListAdapter
+    @Inject lateinit var factory: DealListViewModelFactory
+
+    private var noInternetSnackbar: Snackbar? = null
+    private val onDestroyDisposable = CompositeDisposable()
+    private val onStoreFilterClosed = PublishSubject.create<Unit>()
+
+    private val binding by lazy { DataBindingUtil.setContentView<ActivityDealListBinding>(this, R.layout.activity_deal_list) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // TODO - remember user default choice for deal sorting
+
+        binding.toolbar.inflateMenu(R.menu.deal_list)
+
+        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
+        val searchView = MenuItemCompat.getActionView(binding.toolbar.menu.findItem(R.id.menu_search_games)) as SearchView
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName))
+
+        binding.dealsRecyclerView.adapter = adapter
+
+        val viewModel = ViewModelProviders.of(this, factory).get(DealListViewModel::class.java)
+
+        listOf(
+            binding.dealSortTypeSpinner.itemSelections().map { DealListEvent.SortingChanged(it) }.skip(1),
+            binding.toolbar.itemClicks().map { DealListEvent.MenuItemClicked(it.itemId) },
+            onStoreFilterClosed.map { DealListEvent.StoreFilterClosed }
+            )
+            .merge()
+            .compose(viewModel.eventToUiState)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { (dealViewEntries, showInternetOffMessage, inProgress, hasError, navigation) ->
+                binding.dealsLoadingProgressBar.visible = inProgress
+                binding.dealsRecyclerView.visible = !inProgress
+
+                if (!inProgress && !hasError) {
+                    adapter.viewModels = dealViewEntries
+                }
+
+                if (showInternetOffMessage) showNoInternetMessage() else hideNoInternetMessage()
+
+                when (navigation) {
+                    DealListUiState.Navigation.WATCH_LIST -> openWatchList()
+                    DealListUiState.Navigation.STORE_FILTER -> openStoresFilter()
+                    DealListUiState.Navigation.RATE_APP -> goToStoreAndRate()
+                    DealListUiState.Navigation.SHARE_APP -> showShareAppDialog()
+                    DealListUiState.Navigation.SEND_FEEDBACK -> sendFeedbackEmail()
+                }
+
+            }.addTo(onDestroyDisposable)
```

Comment:
I'm getting `hasError = true` but you don't seem to handle it, so I just get a blank screen.